sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1562,7 +1564,7 @@ def __init__(self,
             self.cluster_name = cluster_name
             self._cluster_yaml = cluster_yaml.replace(os.path.expanduser('~'),
                                                       '~', 1)
-            self.head_ip = head_ip
+            self.stable_internal_external_ips = stable_internal_external_ips
```

Comment:
I would prefer to have this `stable_internal_external_ips` private, and only expose the `interal_ips` and `external_ips` as properties of this class, to make the code more readable and easier to maintain.